### PR TITLE
Refactor according to `geom` changes for slippy tiles

### PR DESCRIPTION
**Pull request by ear7h**
The following PR introduced api changes to slippy tiles, this PR updates the `geom` dependency and refactors accordingly.
https://github.com/go-spatial/geom/pull/22

### CODE REVIEW

File: cmd/tegola/cmd/cache.go [hunk 1]
```diff
@@ -234,9 +234,10 @@ func SeedWorker(ctx context.Context, mt MapTile) error {
 		}
 	}
 
+	// REVIEW (@ear7h:@gdey,@arolek): is this necessary? since tile buffer is set within the map package
 	//	set tile buffer if it was configured by the user
 	if conf.TileBuffer != nil {
-		mt.Tile.Buffer = float64(*conf.TileBuffer)
+		m.TileBuffer = uint64(*conf.TileBuffer)
```
Comment (gdey) [inline cmd/tegola/cmd/cache.go]: @ARolek Do you recall if this should be set on the MapTile or the Map?

File: provider/gpkg/gpkg_test.go [hunk 2]
```diff
@@ -366,14 +352,15 @@ func TestTileFeatures(t *testing.T) {
...
-			},
+			tile: slippy.NewTile(0, 0, 0),
+			//MockTile{
+			//	Z:    0,
+			//	srid: tegola.WebMercator,
+			//	bufferedExtent: geom.NewExtent(
+			//		[2]float64{-20026376.39, -20048966.10},
+			//		[2]float64{20026376.39, 20048966.10},
+			//	),
+			//},
```
Comment (gdey) [inline provider/gpkg/gpkg_test.go]: Please remove this code if it no longer needed. If it is needed, put a comment on top as to the reason why.

File: provider/gpkg/gpkg_test.go [hunk 3]
```diff
@@ -489,13 +465,14 @@ func TestConfigs(t *testing.T) {
...
-			},
+			tile: slippy.NewTile(0, 0, 0),
+			//tile: MockTile{
+			//	bufferedExtent: geom.NewExtent(
+			//		[2]float64{-20026376.39, -20048966.10},
+			//		[2]float64{20026376.39, 20048966.10},
+			//	),
+			//	srid: tegola.WebMercator,
+			//},
 			layerName: "rd_lines",
```
Comment (gdey) [inline provider/gpkg/gpkg_test.go]: If we don’t need this comment block we should remove it.

File: provider/gpkg/gpkg_test.go [hunk 4]
```diff
@@ -457,13 +433,13 @@ func TestConfigs(t *testing.T) {
...
-				bufferedExtent: geom.NewExtent(
-					[2]float64{-20026376.39, -20048966.10},
-					[2]float64{20026376.39, 20048966.10},
-				),
-				srid: tegola.WebMercator,
-			},
+			tile: slippy.NewTile(0, 0, 0),
+				//bufferedExtent: geom.NewExtent(
+				//	[2]float64{-20026376.39, -20048966.10},
+				//	[2]float64{20026376.39, 20048966.10},
```
Comment (gdey) [inline provider/gpkg/gpkg_test.go]: If we don’t need this commented out code block please remove it.

**Comment by ear7h** [comment]
Moved to #508

**Pull request closed without merging**